rasa/core/nlg/response.py

```diff
@@ -25,28 +25,58 @@ def __init__(self, responses: Dict[Text, List[Dict[Text, Any]]]) -> None:
         """
         self.responses = responses
 
+    def _matches_filled_slots(
+        self, filled_slots: Dict[Text, Any], response: Dict[Text, Any],
+    ) -> bool:
+        """Checks if the conditional response variation matches the filled slots."""
+        satisfied_constraints = []
+        constraints = response.get("condition")
+        for constraint in constraints:
+            name = constraint["name"]
+            value = constraint["value"]
+            if filled_slots.get(name) == value:
+                satisfied_constraints.append(constraint)
+
+        if len(satisfied_constraints) == len(constraints):
+            return True
+
+        return False
+
     def _responses_for_utter_action(
-        self, utter_action: Text, output_channel: Text
+        self, utter_action: Text, output_channel: Text, filled_slots: Dict[Text, Any]
     ) -> List[Dict[Text, Any]]:
-        """Return array of responses that fit the channel and action."""
-        channel_responses = []
+        """Return array of responses that fit the channel, action and condition."""
```

Comment:
according to our [code conventions](https://www.notion.so/rasa/Python-Code-Conventions-26835ffd72484dd699f28fa693916d2e#30a6eae49f8f4b4f97c5cf39b577a516) we use "descriptive" docstrings (third person instead of imperative)
```suggestion
        """Returns array of responses that fit the channel, action and condition."""
```